auction-server/src/auction.rs

```diff
@@ -895,15 +906,73 @@ pub fn verify_submit_bid_instruction_svm(
                 .data
                 .starts_with(&express_relay_svm::instruction::SubmitBid::discriminator())
         })
-        .count()
-        != 1
-    {
-        return Err(RestError::BadParameters(
+        .cloned()
+        .collect();
+
+    match submit_bid_instructions.len() {
+        1 => Ok(submit_bid_instructions[0].clone()),
+        _ => Err(RestError::BadParameters(
             "Bid has to include exactly one submit_bid instruction to Express Relay program"
                 .to_string(),
+        )),
+    }
+}
+
+fn extract_account_svm(
+    accounts: &[Pubkey],
+    instruction: CompiledInstruction,
+    position: usize,
+) -> Result<Pubkey, RestError> {
+    if instruction.accounts.len() <= position {
+        tracing::error!(
+            "Account position not found in instruction: {:?} - {}",
+            instruction,
+            position,
+        );
+        return Err(RestError::BadParameters(
+            "Permission account not found in submit_bid instruction".to_string(),
         ));
     }
-    Ok(())
+
+    let account_position = instruction.accounts[position] as usize;
```

Comment:
I think using `into` is safer than `as`:
`let account_position: usize = instruction.accounts[position].into();`